flink-core/src/main/java/org/apache/flink/core/execution/Executor.java

```diff
@@ -18,20 +18,25 @@
 
 package org.apache.flink.core.execution;
 
+import org.apache.flink.annotation.Internal;
 import org.apache.flink.api.common.JobExecutionResult;
 import org.apache.flink.api.dag.Pipeline;
 import org.apache.flink.configuration.Configuration;
 
+import javax.annotation.Nonnull;
+
 /**
  * The entity responsible for executing a {@link Pipeline}, i.e. a user job.
  */
+@Internal
 public interface Executor {
 
 	/**
 	 * Executes a {@link Pipeline} based on the provided configuration.
+	 *
 	 * @param pipeline the {@link Pipeline} to execute
 	 * @param configuration the {@link Configuration} with the required execution parameters
 	 * @return the {@link JobExecutionResult} corresponding to the pipeline execution.
 	 */
-	JobExecutionResult execute(Pipeline pipeline, Configuration configuration) throws Exception;
+	JobExecutionResult execute(@Nonnull final Pipeline pipeline, @Nonnull final Configuration configuration) throws Exception;
```

Comment:
The same here for non-null. Besides, marking a parameter of interface method as `final` means nothing because `final` modifier isn't inherited, but I respect it is expressive at some level. ref [here](https://stackoverflow.com/questions/5380177/final-arguments-in-interface-methods-whats-the-point).

Reply:
the same for non-null in the rest of this commit.

Reply:
I know that we enforce nothing with marking it as `final`, we just express the intentions of the developer that introduced the interface.